code/modules/mod/modules/modules_maint.dm

```diff
@@ -0,0 +1,247 @@
+	SIGNAL_HANDLER
+	balloon_alert(user, "springlocks aren't responding...?")
+	return MOD_CANCEL_ACTIVATE
+
+///Delayed death proc of the suit after the wearer is exposed to reagents
+/obj/item/mod/module/springlock/proc/snap_shut()
+	UnregisterSignal(mod, COMSIG_MOD_ACTIVATE)
+	if(!mod.wearer) //while there is a guaranteed user when on_wearer_exposed() fires, that isn't the same case for this proc
+		return
+	mod.wearer.visible_message("[src] inside [mod.wearer]'s [mod.name] snaps shut, mutilating the user inside!", span_userdanger("*SNAP*"))
+	mod.wearer.emote("scream")
+	playsound(mod.wearer, 'sound/effects/snap.ogg', 75, TRUE, frequency = 0.5)
+	playsound(mod.wearer, 'sound/effects/splat.ogg', 50, TRUE, frequency = 0.5)
+	mod.wearer.apply_damage(500, BRUTE, sharpness = SHARP_POINTY, wound_bonus = -50) //boggers, bogchamp, etc
+	mod.wearer.death() //just in case, for some reason, they're still alive
+	flash_color(mod.wearer, flash_color = "#FF0000", flash_time = 10 SECONDS)
+
+///Rave Visor - Gives you a rainbow visor and plays jukebox music to you.
+/obj/item/mod/module/visor/rave
+	name = "MOD rave visor module"
+	desc = "A Super Cool Awesome Visor (SCAV), intended for modular suits."
+	icon_state = "rave_visor"
+	complexity = 1
+	overlay_state_inactive = "module_rave"
+	var/datum/client_colour/rave_screen
+	var/rave_number = 1
+	var/datum/track/selection
+	var/list/songs = list()
+	var/static/list/rainbow_order = list(
+		"#FF6666",
+		"#FFAA66",
+		"#FFFF66",
+		"#66FF66",
+		"#66AAFF",
+		"#AA66FF",
+		)
+
+/obj/item/mod/module/visor/rave/Initialize(mapload)
+	. = ..()
+	var/list/tracks = flist("[global.config.directory]/jukebox_music/sounds/")
+	for(var/sound in tracks)
+		var/datum/track/track = new()
+		track.song_path = file("[global.config.directory]/jukebox_music/sounds/[sound]")
+		var/list/sound_params = splittext(sound,"+")
+		if(length(sound_params) != 3)
+			continue
+		track.song_name = sound_params[1]
+		track.song_length = text2num(sound_params[2])
+		track.song_beat = text2num(sound_params[3])
+		songs[track.song_name] = track
+	if(length(songs))
+		var/song_name = pick(songs)
+		selection = songs[song_name]
+
+/obj/item/mod/module/visor/rave/on_activation()
+	. = ..()
+	if(!.)
+		return
+	rave_screen = mod.wearer.add_client_colour(/datum/client_colour/rave)
+	rave_screen.update_colour(rainbow_order[rave_number])
+	if(selection)
+		mod.wearer.playsound_local(get_turf(src), null, 50, channel = CHANNEL_JUKEBOX, S = sound(selection.song_path), use_reverb = FALSE)
+
+/obj/item/mod/module/visor/rave/on_deactivation()
+	. = ..()
+	if(!.)
+		return
+	QDEL_NULL(rave_screen)
+	if(selection)
+		mod.wearer.stop_sound_channel(CHANNEL_JUKEBOX)
+
+/obj/item/mod/module/visor/rave/generate_worn_overlay(mutable_appearance/standing)
+	. = ..()
+	for(var/mutable_appearance/appearance as anything in .)
+		appearance.color = active ? rainbow_order[rave_number] : null
```

Comment:
Does it only have one type of visor appearance it can apply, or does it somehow magically apply to every visor?

Reply:
it has one visor that should cover probably like 90% of the sprites perfectly